java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1344,6 +1355,41 @@ public ColumnBuilder appendUTF8String(byte[] value, int srcOffset, int length) {
       return this;
     }
 
+    /**
+     * Append multiple non-null byte values.
+     */
+    public ColumnBuilder append(byte[] value, int srcOffset, int length) {
+      assert type.isBackedByByte();
+      assert srcOffset >= 0;
+      assert length >= 0;
+      assert value.length + srcOffset <= length;
```

Comment:
Should this be: `assert srcOffset + length <= value.length` ?

Reply:
Yes (face palm) I copied and pasted that bug from another part of the code, that I think I wrote a long time ago. Thanks for catching it.

Reply:
One question I'm thinking about most of the time with such code is: whether we should use `if(...) throw new exception()` or `assert`?

Reply:
We try to favor asserts for things that actually touch the data because it can reduce the overhead. The asserts are enabled during tests so we can verify that things are working as expected, but disabled in production so we don't have the performance impact.